pkg/docker/docker_api.go

```diff
@@ -0,0 +1,283 @@
+package docker_api
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"os/user"
+	"path/filepath"
+	"time"
+
+	"github.com/appscode/analytics/pkg/spreadsheet"
+	"github.com/appscode/errors"
+	"golang.org/x/net/context"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/sheets/v4"
+)
+
+type DockerRepositories struct {
+	Count    int         `json:"count"`
+	Next     interface{} `json:"next"`
+	Previous interface{} `json:"previous"`
+	Results  []struct {
+		User           string      `json:"user"`
+		Name           string      `json:"name"`
+		Namespace      string      `json:"namespace"`
+		RepositoryType string      `json:"repository_type"`
+		Status         int         `json:"status"`
+		Description    string      `json:"description"`
+		IsPrivate      bool        `json:"is_private"`
+		IsAutomated    bool        `json:"is_automated"`
+		CanEdit        bool        `json:"can_edit"`
+		StarCount      int         `json:"star_count"`
+		PullCount      int         `json:"pull_count"`
+		LastUpdated    time.Time   `json:"last_updated"`
+		BuildOnCloud   interface{} `json:"build_on_cloud"`
+	} `json:"results"`
+}
+
+type DockerRepoLogs struct {
+	User        string
+	Name        string
+	StarCount   int
+	PullCount   int
+	LastUpdated time.Time
+}
+
+const SpreadSheetIdKubeDB = "10OGrTJxEDox4VR15U7HPGRjiFpTNfhiUGMuL7BQJKJU"   //https://docs.google.com/spreadsheets/d/10OGrTJxEDox4VR15U7HPGRjiFpTNfhiUGMuL7BQJKJU
+const SpreadSheetIdAppsCode = "18lNbYqiP4gsKBoLDoUw2ejOGWN3t3mUOyKGaKeye3kI" //https://docs.google.com/spreadsheets/d/18lNbYqiP4gsKBoLDoUw2ejOGWN3t3mUOyKGaKeye3kI
+
+func getDockerLogs(link string) (*DockerRepositories, error) {
+
+	url := link
+
+	// Build the request
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	client := &http.Client{}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	// Callers should close resp.Body
+	// when done reading from it
+	// Defer the closing of the body
+	defer resp.Body.Close()
+
+	// Fill the record with the data from the JSON
+	var record DockerRepositories
+
+	// Use json.Decode for reading streams of JSON data
+	if err := json.NewDecoder(resp.Body).Decode(&record); err != nil {
+		return nil, err
+	}
+
+	return &record, nil
+}
+
+func updateSheet(dockLogs DockerRepoLogs, SpreadSheetId string) (string, error) {
+	// Usase Limits: https://developers.google.com/sheets/api/limits
+	// 100 Requeste per 100Seconds per User
+	// 500 requests per 100 seconds per project
+	// So, making it slow each request
+	time.Sleep(5 * time.Second)
```

Comment:
time.Sleep doesn't make it slow.
if you are careful about rate limit, then you should call this method when ever it is ok to call.
like call this method every 5 seconds and update data.